### PR TITLE
Consolidate and refactor to single asker/console

### PR DESCRIPTION
**Pull request by wbreza**
Merges the old asker Fn implementation with the new `input.Console` interface to support easier testing and usage with standard return values rather than forcing function parameters with pointers.

### CODE REVIEW

File: cli/azd/cmd/monitor.go [hunk 1]
```diff
@@ -57,7 +58,7 @@ func (m *monitorAction) SetupFlags(
 
 func (m *monitorAction) Run(ctx context.Context, _ *cobra.Command, args []string, azdCtx *environment.AzdContext) error {
 	azCli := commands.GetAzCliFromContext(ctx)
-	askOne := makeAskOne(m.rootOptions.NoPrompt)
+	console := input.NewAskerConsole(!m.rootOptions.NoPrompt)
```
Comment (vhvb1989) [inline cli/azd/cmd/monitor.go:61]: consider keeping the same input-arg logic for `NewAskerConsole` so we don't need to add the negation all the times (`!`)

File: cli/azd/cmd/env.go [hunk 2]
```diff
@@ -54,7 +55,7 @@ You can find all environment configurations under the *.azure\<environment-name>
 
 func envSetCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
 	actionFn := func(ctx context.Context, _ *cobra.Command, args []string, azdCtx *environment.AzdContext) error {
-		askOne := makeAskOne(rootOptions.NoPrompt)
+		console := input.NewAskerConsole(!rootOptions.NoPrompt)
```
Comment (weikanglim) [inline cli/azd/cmd/env.go:58]: I suspect the `Console` interface we've introduced here is for mocking. We don't actually expect multiple implementations of `Console` out in production code.

I wonder if we should just rename `NewAskerConsole` as `NewConsole` for improved readability.

File: cli/azd/cmd/monitor.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/cmd/monitor.go:61]: I like Wallace's direction of `Interactive` over `NoPrompt`. I think we could avoid the negation for the current change, and as a separate change, refactor all usages of `NoPrompt` (including in `rootOptions`) -> `Interactive`. That way we avoid the bool conversions.

File: cli/azd/pkg/input/console.go [hunk 3]
```diff
@@ -102,18 +103,19 @@ func (c *AskerConsole) Confirm(ctx context.Context, options ConsoleOptions) (boo
 
 // PromptTemplate ask the user to select a template.
 // An empty string is returned if the user selects 'Empty Template' from the choices
```
Comment (weikanglim) [inline cli/azd/pkg/input/console.go:105]: Comment is outdated

File: cli/azd/pkg/input/console.go [hunk 4]
```diff
@@ -102,18 +103,19 @@ func (c *AskerConsole) Confirm(ctx context.Context, options ConsoleOptions) (boo
...
-func (c *AskerConsole) PromptTemplate(ctx context.Context, message string) (string, error) {
+func (c *AskerConsole) PromptTemplate(ctx context.Context, message string) (templates.Template, error) {
+	var result templates.Template
```
Comment (weikanglim) [inline cli/azd/pkg/input/console.go:107]: Seems like we could remove this and return `nil` explicitly instead in error case.

**Review by vhvb1989 — APPROVED** [review]

File: cli/azd/cmd/env.go (hunk 2 above)
Comment (wbreza) [reply cli/azd/cmd/env.go:58]: Correct - the interface is to help aid in mocking/testing.  I will rename creation method to `NewConsole`.

File: cli/azd/pkg/input/console.go (hunk 4 above)
Comment (wbreza) [reply cli/azd/pkg/input/console.go:107]: Going to hold off on changing the contract and keep as `(Template, error)`

**Pull request merged**